posydon/binary_evol/DT/step_detached.py

```diff
@@ -1217,7 +1217,7 @@ def ev_rlo1(t, y):
             if (interp1d_sec["mass"](t - t_offset_sec) == 0) or (interp1d_pri["mass"](t - t_offset_sec) == 0):
                 warnings.warn("Trying to compute difference between stellar radii for non-existent companion",
                             EvolutionWarning)
-                RL = np.nan
+                RL = np.ones_like(interp1d_sec["mass"](t - t_offset_sec))*np.nan
```

Comment:
What is the attempt here? Conserving the shape of the interpolator output? If so, why the one of the secondary and not the one of the primary?
I think, we shouldn't infer a shape here, the doc string says, we are returning a float not a numpy array.
So instead we should better check that the interpolator outputs are floats otherwise we should raise an error.
Additionally, I don't know, why we check for 0 here, we should disallow any mass <=0.

P.S.: This is a bad use of np.ones_like(array)! Better use np.fill_like(array, fill_value). This does the same as you do with the multiplication, but in one step instead of two. Btw. np.fill_like(array, 1.0) is the same as np.ones_like(array) in the background code; hence I don't see any good use case for np.ones_like. My guess is that np.ones_like might existed first with a different implementation, but these days I'd call it outdated.

Reply:
I was finding that simply setting RL = np.nan was causing problems because while the function says it returns a float, in reality it is returning an odd data structure containing one float value. I assumed that the primary and secondary interpolator output would have the same shape?
Will look at this again.

Reply:
<img width="298" alt="Screen Shot 2024-07-10 at 10 01 01 AM" src="https://github.com/POSYDON-code/POSYDON/assets/13874913/87d2a49a-91e7-46ff-84ad-a51616f08db1">
@mkruckow I printed-out the data types of the RL radius being calculated in the original code (not my changes) and you can see that sometimes it has a type of numpy float64 and other times it is an ndarray (presumably because the type of the interpolator output is changing). If you think this is a bigger issue, I can just leave this alone for now and revert back to the original code.

Reply:
Hm, if we want to support numpy arrays, we should update the doc string, ensure that all the arrays have the same shape, and switch to use `np.full_like`.
We need to update the check for masses, too. Because this might not work, in case there is a 0 inside the numpy array. Additionally, we should exclude negative masses in this go, too.

Reply:
@celiotine Do you know why it returns a numpy array?
Is it just a single value inside an array? Or are there multiple values?

Reply:
I do not know why the interpolator is returning a numpy array for some values. From what I have checked, it contains one float value but has an odd shape?

Reply:
BTW I don't really have an interest in rewriting this function, I was just trying to turn a divide-by-zero warning into a POSYDON warning. If there are bigger issues here, I think that should be in a different PR

Reply:
No worries! I agree with not rewritting the function as part of this PR :) 
Maybe make is a separate issue instead?

As for this code, does the `roche_lobe_radius` function crash if a non-existent companion calculation is done or is it just giving the divide-by-zero warning? If the latter, you could just "suppress" the divide-by-zero warning for this part of the code, since you're throwing the POSYDON error, and still call the `roche_lobe_radius` function. Then you do not have to deal with proper handling of the Roche Lobe function input/output ;)  

```
with np.errstate(divide='ignore'):
    RL = roche_lobe_radius(....)
```

or if it's 0/0:

```
with np.errstate(divide='ignore', invalid='ignore'):
    RL = roche_lobe_radius(....)
```

Reply:
Yes that is what I was thinking! The code doesn't crash, it just throws the python warning. I might be able to suppress the warning locally as an easy fix, and create an issue for the interpolator output/this function in general. Thanks!!